### ISSUE TITLE
Targets with non-ASCII filenames as inputs are always invalidated on Windows

### ISSUE THREAD

**Issue by svraka** [ropensci/targets#814]
## Prework

* [X] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [X] Confirm that your issue is a genuine bug in the `targets` package itself and not a user error, known limitation, or issue from another package that `targets` depends on. For example, if you get errors running `tar_make_clustermq()`, try isolating the problem in a reproducible example that runs `clustermq` and not `targets`. And for miscellaneous troubleshooting, please post to [discussions](https://github.com/ropensci/targets/discussions) instead of [issues](https://github.com/ropensci/targets/issues).
* [X] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [X] Using [`targets::tar_reprex()`](https://docs.ropensci.org/targets/reference/tar_reprex.html), [`reprex::reprex()`](https://reprex.tidyverse.org/reference/reprex.html), or similar, post a [minimal reproducible example](https://www.tidyverse.org/help/) like [this one](https://github.com/ropensci/targets/issues/256#issuecomment-754229683) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [X] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [X] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [X] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

Consider a pipeline where a target depends on a file with non-ASCII filename. Following examples and advice from related issues (#249, #633) it works as expected on Linux but on Windows the file target is invalidated on every run.

## Reproducible example

* [X] Using [`targets::tar_reprex()`](https://docs.ropensci.org/targets/reference/tar_reprex.html), [`reprex::reprex()`](https://reprex.tidyverse.org/reference/reprex.html), or similar, post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [X] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [X] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [X] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

Here's the reprex with some diagnostics at the end on Linux:

``` r
options(encoding = "UTF-8")
targets::tar_dir({
  library(targets)
  writeLines("foo", "á.csv")
  writeLines("options(encoding = \"UTF-8\")", ".Rprofile")
  tar_script({
    library(targets)
    list(tar_target(file, "á.csv", format = "file"),
         tar_target(content, readLines(file)))
  }, ask = FALSE)
  tar_make()
  cat("outdated:", tar_outdated(), "\n")
  print(tar_read(file))
  print(tar_read(content))
})
#> • start target file
#> • built target file
#> • start target content
#> • built target content
#> • end pipeline: 0.085 seconds
#> outdated:  
#> [1] "á.csv"
#> [1] "foo"
```

<sup>Created on 2022-03-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

And on Windows:

``` r
options(encoding = "UTF-8")
targets::tar_dir({
  library(targets)
  writeLines("foo", "á.csv")
  writeLines("options(encoding = \"UTF-8\")", ".Rprofile")
  tar_script({
    library(targets)
    list(tar_target(file, "á.csv", format = "file"),
         tar_target(content, readLines(file)))
  }, ask = FALSE)
  tar_make()
  cat("outdated:", tar_outdated(), "\n")
  print(tar_read(file))
  print(tar_read(content))
})
#> * start target file
#> * built target file
#> * start target content
#> * built target content
#> * end pipeline: 0.09 seconds
#> outdated: content file 
#> [1] "Ă\241.csv"
#> [1] "foo"
```

<sup>Created on 2022-03-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

The input file is found, processed but it seems, it's not stored correctly, as reading the path with `tar_read()` results in mojibake on Windows.

## Expected result

Targets are not invalidated.

## Diagnostic information

<details style="margin-bottom:10px;">
<summary>
Session info, Linux
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.1.3 (2022-03-10)
#>  os       Ubuntu 20.04.4 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       Etc/UTC
#>  date     2022-03-23
#>  pandoc   2.17.1.1 @ /home/linuxbrew/.linuxbrew/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  backports     1.4.1   2021-12-13 [1] RSPM (R 4.1.0)
#>  base64url     1.4     2018-05-14 [1] RSPM (R 4.1.0)
#>  callr         3.7.0   2021-04-20 [1] RSPM (R 4.0.4)
#>  cli           3.2.0   2022-02-14 [1] RSPM (R 4.1.0)
#>  codetools     0.2-18  2020-11-04 [3] CRAN (R 4.0.3)
#>  crayon        1.5.0   2022-02-14 [1] RSPM (R 4.1.0)
#>  data.table    1.14.2  2021-09-27 [1] CRAN (R 4.1.1)
#>  digest        0.6.29  2021-12-01 [1] RSPM (R 4.1.0)
#>  ellipsis      0.3.2   2021-04-29 [1] RSPM (R 4.0.4)
#>  evaluate      0.15    2022-02-18 [1] RSPM (R 4.1.0)
#>  fansi         1.0.2   2022-01-14 [1] CRAN (R 4.1.2)
#>  fastmap       1.1.0   2021-01-25 [1] RSPM (R 4.0.3)
#>  fs            1.5.2   2021-12-08 [1] CRAN (R 4.1.2)
#>  glue          1.6.2   2022-02-24 [1] RSPM (R 4.1.0)
#>  highr         0.9     2021-04-16 [1] CRAN (R 4.0.5)
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.1.1)
#>  igraph        1.2.11  2022-01-04 [1] RSPM (R 4.1.0)
#>  knitr         1.37    2021-12-16 [1] CRAN (R 4.1.2)
#>  lifecycle     1.0.1   2021-09-24 [1] RSPM (R 4.1.0)
#>  magrittr      2.0.2   2022-01-26 [1] RSPM (R 4.1.0)
#>  pillar        1.7.0   2022-02-01 [1] CRAN (R 4.1.2)
#>  pkgconfig     2.0.3   2019-09-22 [1] RSPM (R 4.0.0)
#>  processx      3.5.2   2021-04-30 [1] RSPM (R 4.0.4)
#>  ps            1.6.0   2021-02-28 [1] RSPM (R 4.0.3)
#>  purrr         0.3.4   2020-04-17 [1] RSPM (R 4.0.0)
#>  R6            2.5.1   2021-08-19 [1] RSPM (R 4.1.0)
#>  reprex        2.0.1   2021-08-05 [1] CRAN (R 4.1.0)
#>  rlang         1.0.2   2022-03-04 [1] CRAN (R 4.1.2)
#>  rmarkdown     2.13    2022-03-10 [1] RSPM (R 4.1.0)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.1.2)
#>  stringi       1.7.6   2021-11-29 [1] CRAN (R 4.1.2)
#>  stringr       1.4.0   2019-02-10 [1] RSPM (R 4.0.0)
#>  targets     * 0.11.0  2022-03-19 [1] RSPM (R 4.1.0)
#>  tibble        3.1.6   2021-11-07 [1] CRAN (R 4.1.2)
#>  tidyselect    1.1.2   2022-02-21 [1] RSPM (R 4.1.0)
#>  utf8          1.2.2   2021-07-24 [1] RSPM (R 4.1.0)
#>  vctrs         0.3.8   2021-04-29 [1] RSPM (R 4.0.4)
#>  withr         2.5.0   2022-03-03 [1] RSPM (R 4.1.0)
#>  xfun          0.30    2022-03-02 [1] RSPM (R 4.1.0)
#>  yaml          2.3.5   2022-02-21 [1] RSPM (R 4.1.0)
#> 
#>  [1] /usr/local/lib/R/site-library
#>  [2] /usr/lib/R/site-library
#>  [3] /usr/lib/R/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

<details style="margin-bottom:10px;">
<summary>
Session info, Windows
</summary>

``` r
sessioninfo::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value
#>  version  R version 4.1.3 (2022-03-10)
#>  os       Windows 10 x64 (build 18363)
#>  system   x86_64, mingw32
#>  ui       RTerm
#>  language (EN)
#>  collate  Hungarian_Hungary.1250
#>  ctype    Hungarian_Hungary.1250
#>  tz       Europe/Prague
#>  date     2022-03-23
#>  pandoc   2.17.1.1 @ C:/scoop/shims/ (via rmarkdown)
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version date (UTC) lib source
#>  backports     1.4.1   2021-12-13 [1] CRAN (R 4.1.2)
#>  base64url     1.4     2018-05-14 [1] CRAN (R 4.1.3)
#>  callr         3.7.0   2021-04-20 [1] CRAN (R 4.1.3)
#>  cli           3.2.0   2022-02-14 [1] CRAN (R 4.1.3)
#>  codetools     0.2-18  2020-11-04 [1] CRAN (R 4.1.3)
#>  crayon        1.5.0   2022-02-14 [1] CRAN (R 4.1.3)
#>  data.table    1.14.2  2021-09-27 [1] CRAN (R 4.1.3)
#>  digest        0.6.29  2021-12-01 [1] CRAN (R 4.1.3)
#>  ellipsis      0.3.2   2021-04-29 [1] CRAN (R 4.1.3)
#>  evaluate      0.15    2022-02-18 [1] CRAN (R 4.1.3)
#>  fansi         1.0.2   2022-01-14 [1] CRAN (R 4.1.3)
#>  fastmap       1.1.0   2021-01-25 [1] CRAN (R 4.1.3)
#>  fs            1.5.2   2021-12-08 [1] CRAN (R 4.1.3)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.1.3)
#>  highr         0.9     2021-04-16 [1] CRAN (R 4.1.3)
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.1.3)
#>  igraph        1.2.11  2022-01-04 [1] CRAN (R 4.1.3)
#>  knitr         1.37    2021-12-16 [1] CRAN (R 4.1.3)
#>  lifecycle     1.0.1   2021-09-24 [1] CRAN (R 4.1.3)
#>  magrittr      2.0.2   2022-01-26 [1] CRAN (R 4.1.3)
#>  pillar        1.7.0   2022-02-01 [1] CRAN (R 4.1.3)
#>  pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.1.3)
#>  processx      3.5.2   2021-04-30 [1] CRAN (R 4.1.3)
#>  ps            1.6.0   2021-02-28 [1] CRAN (R 4.1.3)
#>  purrr         0.3.4   2020-04-17 [1] CRAN (R 4.1.3)
#>  R6            2.5.1   2021-08-19 [1] CRAN (R 4.1.3)
#>  reprex        2.0.1   2021-08-05 [1] CRAN (R 4.1.3)
#>  rlang         1.0.2   2022-03-04 [1] CRAN (R 4.1.3)
#>  rmarkdown     2.13    2022-03-10 [1] CRAN (R 4.1.3)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.1.3)
#>  stringi       1.7.6   2021-11-29 [1] CRAN (R 4.1.2)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.1.3)
#>  targets     * 0.11.0  2022-03-19 [1] CRAN (R 4.1.3)
#>  tibble        3.1.6   2021-11-07 [1] CRAN (R 4.1.3)
#>  tidyselect    1.1.2   2022-02-21 [1] CRAN (R 4.1.3)
#>  utf8          1.2.2   2021-07-24 [1] CRAN (R 4.1.3)
#>  vctrs         0.3.8   2021-04-29 [1] CRAN (R 4.1.3)
#>  withr         2.5.0   2022-03-03 [1] CRAN (R 4.1.3)
#>  xfun          0.30    2022-03-02 [1] CRAN (R 4.1.3)
#>  yaml          2.3.5   2022-02-21 [1] CRAN (R 4.1.2)
#> 
#>  [1] C:/scoop/apps/r-release/4.1.3/library
#> 
#> ------------------------------------------------------------------------------
```

</details>

**Comment by svraka** [comment]
Thanks for the quick reply, it does work. 

Just to note, be careful where you set `options(encoding = "UTF-8")` because running reprexes in an interactive session with UTF-8 breaks lots of things, including this fix 😄 Hopefully, this might go away with R 4.2.

**Comment by wlandau** [comment]
Thanks for confirming.
